### PR TITLE
Day 2 cpp. Some ranges/view/split gymnastics.

### PR DESCRIPTION
**Pull request by RongkunWang**
clearly Mac clang and g++ on github has some differences that some functions is only available in clang..

Could be used as an example for future parsing.

### CODE REVIEW

**Comment by Moelf** [comment]
yeah I don't have the best idea of setting up C++ CI, from what I can tell, we at least have two axes of options...

## Compiler technology:
- g++
- Clang

## Build system:
- make/CMake
- Bezel
- ...


maybe #8 should split the content of `src/` folder into `cpp_bezel/`

File: runtests.jl [hunk 1]
```diff
@@ -18,7 +18,7 @@ function run_solution() end
 
 function run_solution(source_path, input_path, ::Val{:cpp})
     exename = replace(splitpath(source_path)[end], "cpp" => "o")
-    run(`g++ -std=c++20 $source_path -o $exename`)
+    run(`g++ -std=c++23 $source_path -o $exename`)
```
Comment (Moelf) [inline runtests.jl]: wow, living in the future

Comment (RongkunWang) [reply runtests.jl]: even c++23 doesn't have the function I wanted, but another way was used to implement the transform lambda function and solved the problem.

Comment (RongkunWang) [reply runtests.jl]: do you have access? https://github.com/RongkunWang/AoC2023/actions/runs/7072796215/job/19252036397#step:6:123

Comment (Moelf) [reply runtests.jl]: yes. that's quite an error...
```cpp
error: no matching function for call to ‘std::ranges::split_view<std::ranges::ref_view<const std::__cxx11::basic_string<char> >, std::ranges::single_view<char> >::_OuterIter<false>::value_type::data()’
   13 |     std::string_view s (v.data(), v.data() + v.size());
      |                         ~~~~~~^~
```

Comment (Cryptum169) [reply runtests.jl]: I'll take a look tonight

Comment (Cryptum169) [reply runtests.jl]: Is this related to https://stackoverflow.com/questions/56118941/do-we-have-c20-ranges-library-in-gcc-9 ? I know it looks like it's just one function and not the whole library missing.

Comment (Moelf) [reply runtests.jl]: Ah, so maybe we need a newer gcc?

**Pull request merged**